source/aggregation.txt

```diff
@@ -0,0 +1,210 @@
+database from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+free MongoDB Atlas cluster and load the sample datasets, see the
+:atlas:`Get Started with Atlas </getting-started>` guide.
+
+The following {+language+} data class models the documents in this collection:
+
+.. literalinclude:: /includes/aggregation/aggregation.kt
+   :start-after: start-data-class
+   :end-before: end-data-class
+   :language: kotlin
+   :copyable:
+
+Build and Execute an Aggregation Pipeline
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To perform an aggregation, pass a list of aggregation stages to the
+``collection.aggregate()`` method.
+
+The following code example produces a count of the number of bakeries in each borough
+of New York. To do so, it uses an aggregation pipeline with the following stages:
+
+- A :manual:`$match </reference/operator/aggregation/match/>` stage to filter for documents
+  whose ``cuisine`` field contains the value ``"Bakery"``.
+
+- A :manual:`$group </reference/operator/aggregation/group/>` stage to group the matching
+  documents by the ``borough`` field, accumulating a count of documents for each distinct
+  value.
+
+.. io-code-block::
+
+   .. input:: /includes/aggregation/aggregation.kt
+      :language: kotlin
+      :start-after: start-aggregation-pipeline
+      :end-before: end-aggregation-pipeline
+      :dedent:
+
+   .. output::
+      :visible: false
+
+      Document{{_id=Bronx, count=71}}
+      Document{{_id=Manhattan, count=221}}
+      Document{{_id=Brooklyn, count=173}}
+      Document{{_id=Queens, count=204}}
+      Document{{_id=Staten Island, count=20}}
+      Document{{_id=Missing, count=2}}
+
+Explain an Aggregation
+~~~~~~~~~~~~~~~~~~~~~~
+
+To view information about how MongoDB executes your operation, you can
+instruct MongoDB to **explain** it. When MongoDB explains an operation, it returns
+**execution plans** and performance statistics. An execution
+plan is a potential way MongoDB can complete an operation.
+When you instruct MongoDB to explain an operation, it returns both the
+plan MongoDB executed and any rejected execution plans.
+
+The following code example runs the preceding aggregation example and prints the returned
+explanation:
+
+.. io-code-block::
+
+   .. input:: /includes/aggregation/aggregation.kt
+      :language: kotlin
+      :start-after: start-aggregation-explain
+      :end-before: end-aggregation-explain
+      :dedent:
+
+   .. output::
+      :visible: false
+    
+      {
+        "explain": {
+          "aggregate": "restaurants",
+          "pipeline": [
+            {
+              "$match": {
+                "cuisine": "Bakery"
+              }
+            },
+            {
+              "$group": {
+                "_id": "$borough",
+                "count": {
+                  "$sum": 1
+                }
+              }
+            }
+          ],
+          "cursor": {}
+        },
+        ...
+      }
```

Comment:
S: it would be helpful if this output included the info about the execution plans etc